sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -2040,6 +2040,23 @@ class DatasetSuite extends QueryTest
       joined,
       (1, 1), (2, 2), (3, 3))
   }
+
+  test("SPARK-36210: withColumns preserve insertion ordering") {
+    val df = Seq(1, 2, 3).toDS()
```

Comment:
can we add a test with a duplicate column names?

Reply:
I think `withColumns`  doesn't allow duplicate column names.

Reply:
There is one test in `DataFrameSuite`.